### ISSUE TITLE
[Bug report] MNIST training loss starts to go up after 3 epochs

### ISSUE THREAD

**Issue by rfurko-tt** [tenstorrent/tt-metal#17555]
After three epochs training of MNIST loss starts to go up. It didn't happen before. It is expected that MNIST should train without training loss going up.
